src/daft-core/src/array/ops/mean.rs

```diff
@@ -39,6 +36,6 @@ impl DaftMeanAggable for &DataArray<Float64Type> {
                 (s, c) => Some(s / (*c as f64)),
             });
         let mean_array = Box::new(PrimitiveArray::from_trusted_len_iter(mean_per_group));
-        Ok(DataArray::from((self.field.name.as_ref(), mean_array)))
+        Ok(Self::from((self.field.name.as_ref(), mean_array)))
```

Comment:
Is there a reason why you're not using `stats` here? I see you used it for ungrouped mean but not grouped.

Reply:
I could use that `grouped_stats` function here. The only "concern" that I had is that it would also need to iterate through the `groups`, even though they're not needed in this grouped-mean implementation.

From a performance perspective, I thought that would be sub-optimal. However, I haven't tested that thought.

Reply:
Can change it over to seem more consistent!

Reply:
Gotcha, I think it's fine to just stick to the existing implementation then!

Reply:
Updated.